server/src/main/java/org/elasticsearch/action/admin/cluster/state/RemoteClusterStateRequest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.action.admin.cluster.state;
+
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.tasks.CancellableTask;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.tasks.TaskId;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * A remote-only version of {@link ClusterStateRequest} that should be used for cross-cluster requests.
+ * It simply exists to handle incoming remote requests and forward them to the local transport action.
+ */
+public class RemoteClusterStateRequest extends ActionRequest {
+
+    private final ClusterStateRequest clusterStateRequest;
+
+    public RemoteClusterStateRequest(ClusterStateRequest clusterStateRequest) {
+        this.clusterStateRequest = clusterStateRequest;
+    }
+
+    public RemoteClusterStateRequest(StreamInput in) throws IOException {
+        this.clusterStateRequest = new ClusterStateRequest(in);
```

Comment:
I'd rather we did the deserialization here to line up with the `writeTo()` implementation, and dropped this constructor from `ClusterStateRequest`.